File: notes/guess/guess.lua

math.randomseed( os.time() )

local from = 1
local to = 100
local rand = math.random( from, to )
local guess = 0
local guesses = 0
 
repeat
  print("Guess the number: ")
  guesses = guesses + 1
  guess = tonumber( io.read() )
  if guess > rand then
    print("Too high!")
  elseif guess < rand then
    print("Too low!")
  else
    print("You got it!")
    print("It took you " .. guesses .. " guesses.")
  end
until guess == rand

File: notes/guess/guess.java

Scanner scan = new Scanner(System.in);
Random random = new Random();
long from = 1;
long to = 100;
int randomNumber = random.nextInt(to - from + 1) + from;
int guessedNumber = 0;
int guesses = 0;

do {
  System.out.print("Guess the number: ");
  guesses++;
  guessedNumber = scan.nextInt();
  if (guessedNumber > randomNumber) {
    System.out.println("Too high!");
  } else if (guessedNumber < randomNumber) {
    System.out.println("Too low!");
  } else {
    System.out.println("You got it!");
    System.out.println("It took you " + guesses + " guesses.");
  }
} while (guessedNumber != randomNumber);